src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/http_action/config/PerformanceAnalyzerClusterConfigActionTests.java

```diff
@@ -0,0 +1,145 @@
+    circuitBreakerService = new HierarchyCircuitBreakerService(Settings.EMPTY, new ArrayList<BreakerSettings>(), clusterSettings);
+    UsageService usageService = new UsageService();
+    threadPool = new TestThreadPool("test");
+    nodeClient = new NodeClient(Settings.EMPTY, threadPool);
+    restController = new RestController(Collections.emptySet(), null, nodeClient, circuitBreakerService, usageService);
+    clusterSettingHandler = new PerformanceAnalyzerClusterSettingHandler(controller, clusterSettingsManager);
+    nodeStatsSettingHandler = new NodeStatsSettingHandler(controller, clusterSettingsManager);
+    configAction = new PerformanceAnalyzerClusterConfigAction(Settings.EMPTY, restController, clusterSettingHandler, nodeStatsSettingHandler);
+    restController.registerHandler(configAction);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    threadPool.shutdownNow();
+  }
+
+  @Test
+  public void testRoutes() {
+    List<Route> routes = configAction.routes();
+    assertEquals(8, routes.size());
+  }
+
+  @Test
+  public void testGetName() {
+    assertEquals(PerformanceAnalyzerClusterConfigAction.class.getSimpleName(), configAction.getName());
+  }
+
+  @Test
+  public void testUpdateRcaSetting() throws IOException {
+    test(PerformanceAnalyzerClusterConfigAction.RCA_CLUSTER_CONFIG_PATH);
+  }
+
+  @Test
+  public void testUpdateLoggingSetting() throws IOException {
+    test(PerformanceAnalyzerClusterConfigAction.LOGGING_CLUSTER_CONFIG_PATH);
+  }
+
+  @Test
+  public void testUpdateBatchMetricsSetting() throws IOException {
+    test(PerformanceAnalyzerClusterConfigAction.BATCH_METRICS_CLUSTER_CONFIG_PATH);
+  }
+
+  @Test
+  public void testUpdatePerformanceAnalyzerSetting() throws IOException {
+    test(PerformanceAnalyzerClusterConfigAction.PA_CLUSTER_CONFIG_PATH);
+  }
+
+  private void test(String requestPath) throws IOException {
+    final FakeRestRequest fakeRestRequest = buildRequest(requestPath);
+    final FakeRestChannel channel = new FakeRestChannel(fakeRestRequest, true, 10);
+    restController.dispatchRequest(fakeRestRequest, channel, new ThreadContext(Settings.EMPTY));
+    assertEquals(RestStatus.OK, channel.capturedResponse().status());
+
+    String responseStr = channel.capturedResponse().content().utf8ToString();
+    assertTrue(responseStr.contains(PerformanceAnalyzerClusterConfigAction.CURRENT));
+    assertTrue(responseStr.contains(PerformanceAnalyzerClusterConfigAction.SHARDS_PER_COLLECTION));
+    assertTrue(responseStr.contains(PerformanceAnalyzerClusterConfigAction.BATCH_METRICS_RETENTION_PERIOD_MINUTES));
+  }
+
+  private FakeRestRequest buildRequest(String requestPath) throws IOException {
+    final XContentBuilder builder = XContentFactory.jsonBuilder()
+        .startObject()
+        .field(PerformanceAnalyzerClusterConfigAction.ENABLED, true)
+        .field(PerformanceAnalyzerClusterConfigAction.SHARDS_PER_COLLECTION, 1)
+        .endObject();
```

Comment:
Let's add indent and curly bracket for any json payload to make it more readable. 
.startObject() {
.field(PerformanceAnalyzerClusterConfigAction.ENABLED, true)
.field(PerformanceAnalyzerClusterConfigAction.SHARDS_PER_COLLECTION, 1)
}
.endObject();

Reply:
good suggestion

Reply:
I'm getting compile error when adding curly bracket, so just add indent for readability. Let me know if you have other suggestions. thanks.